src/ChangeFeedIterator.ts

```diff
@@ -0,0 +1,149 @@
+/// <reference lib="esnext.asynciterable" />
+import { isNumber, isString } from "util";
+import { ChangeFeedOptions } from "./ChangeFeedOptions";
+import { ChangeFeedResponse } from "./ChangeFeedResponse";
+import { Resource } from "./client";
+import { ClientContext } from "./ClientContext";
+import { Constants, ResourceType, StatusCodes } from "./common";
+import { FeedOptions } from "./request";
+import { Response } from "./request";
+
+/**
+ * Provides iterator for change feed.
+ *
+ * Use `Items.readChangeFeed` to get an instance of the iterator.
+ */
+export class ChangeFeedIterator<T> {
```

Comment:
Why do we call this Iterator? It doesn't have the "next" method. As it is, it's more like a ChangeFeedQuery in C# SDK. Like Iterable but without [Symbol.iterator]/[Symbol.asyncIterator] member, and still "query" makes more sense for naming as it has more stuf than just the iterator. Note that I can't use it with for..of. Why not? here's what would be very natural to do in JS:
for await (var item Items.readChangeFeed(options)) process(item);

Reply:
The way we've implemented the async iterator, it is backward compatible (but folks will only be able to use `for await` in the newer versions of Node.js or with babel/typescript polyfills). We already use this async Iterator pattern in the `QueryIterator` type.

Query returns a `QueryIterator`, so it made sense to me that change feed returns a `ChangeFeedIterator`. `ChangeFeedQuery` isn't consistent with our existing naming in the JS SDK.

FWIW, I'd prefer `*Cursor` for both to avoid the `Iterator` confusion, but not confuse SQL queries, read feed with change feed.